### ISSUE TITLE
Unable to use Skaffold + Jib + Docker for Desktop Kubernetes - local image not found

### ISSUE THREAD

**Issue by saturnism** [GoogleContainerTools/skaffold#3834]
Using Skaffold + Jib with Docker for Desktop and the built in Kubernetes locally.

### Expected behavior
My container should start automatically.

### Actual behavior
Application doesn't start due to non-existing image tag.

### Information

- Skaffold version: 1.5.0
- Operating system: macOS Catalina
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta14
kind: Config
build:
  artifacts:
  - image: jib-demo
    jib:
      args:
        - -DskipTests
deploy:
  kubectl:
    manifests: ["k8s/*.yaml"]
```

Skaffold output
```
skaffold dev --filename skaffold.yaml --label ide=idea --label ideVersion=2019.3.1.0.0 --label ijPluginVersion=unknown --rpc-port 50052 --port-forward=true --status-check=true --enable-rpc=true
time="2020-03-16T22:19:15-04:00" level=warning msg="port 50052 for gRPC HTTP server already in use: using 50053 instead"
Listing files to watch...
 - jib-demo
Generating tags...
 - jib-demo -> jib-demo:10cbbae-dirty
Checking cache...
 - jib-demo: Found Locally
Tags used in deployment:
 - jib-demo -> jib-demo:d2f7453839111a590d20c1eaa4ca96364ea5267d024133bdbdcd7eaa49e2ed50
   local images can't be referenced by digest. They are tagged and referenced by a unique ID instead
Starting deploy...
...
Waiting for deployments to stabilize
 - default:deployment/demo Waiting for deployment "demo" rollout to finish: 0 of 1 updated replicas are available...
```

`imagePullPolicy` is already set to `IfNotPresent.

The tag used in deployment is `jib-demo:d2f7453839111a590d20c1eaa4ca96364ea5267d024133bdbdcd7eaa49e2ed50`, which is non-existing.

If I manually switch the tag to `jib-demo:latest`, then it worked well.

`docker images` does not show the image w/ any other tags except for the `latest` tag.

**Comment by saturnism** [comment]
I simply changed to `minikube`, and the error went away :(

```
skaffold dev --filename skaffold.yaml --label ide=idea --label ideVersion=2019.3.1.0.0 --label ijPluginVersion=unknown --rpc-port 50052 --port-forward=true --status-check=true --enable-rpc=true
time="2020-03-16T22:39:43-04:00" level=warning msg="port 50052 for gRPC HTTP server already in use: using 50053 instead"
Listing files to watch...
 - jib-demo
Generating tags...
 - jib-demo -> jib-demo:dd6da0d-dirty
Checking cache...
 - jib-demo: Not found. Building
Found [minikube] context, using local docker daemon.
Building [jib-demo]...
...
```

**Comment by briandealwis** [comment]
If you have a chance to run it again, could you enable debug logging with `-v debug`?

**Comment by dgageot** [comment]
FWIW, `imagePullPolicy` doesn't need to be set. `-v debug` would for sure help understand what's going on. I suspect that the docker daemon used a build time and the one used a deploy time are not the same for whatever reason.

**Comment by saturnism** [comment]
Weirdly enough, I was only able to produce this when running from within Cloud Code, using skaffold 1.5.0 that's in the path. Same issue when running from Cloud Code using cloud Code installed skaffold.

First, make sure images are removed, e.g.
```
docker rmi gcr.io/wise-coyote-827/jib-demo:.......
docker rmi gcr.io/wise-coyote-827/jib-demo:.....-drity
```

When the target image is not already in the Docker Daemon, everything works fine from the Terminal.
```
skaffold dev --filename skaffold.yaml --label ide=idea --label ideVersion=2019.3.1.0.0 --label ijPluginVersion=unknown --rpc-port 50052 --port-forward=true --status-check=true --enable-rpc=true
WARN[0000] port 50052 for gRPC HTTP server already in use: using 50053 instead 
Listing files to watch...
 - gcr.io/wise-coyote-827/jib-demo
Generating tags...
 - gcr.io/wise-coyote-827/jib-demo -> gcr.io/wise-coyote-827/jib-demo:d0be6fa-dirty
Checking cache...
 - gcr.io/wise-coyote-827/jib-demo: Found Locally
Tags used in deployment:
 - gcr.io/wise-coyote-827/jib-demo -> gcr.io/wise-coyote-827/jib-demo:86ffb97a970f634f9d351bf122655f4df59f1f547ab38b1a29b6c50cd1a2efc4
   local images can't be referenced by digest. They are tagged and referenced by a unique ID instead
Starting deploy...
 - service/demo created
 - deployment.apps/demo created
 - secret/demo-postgresql created
 - service/demo-postgresql-headless created
 - service/demo-postgresql created
 - statefulset.apps/demo-postgresql created
Waiting for deployments to stabilize
 - default:deployment/demo Waiting for deployment "demo" rollout to finish: 0 of 1 updated replicas are available...
 - default:deployment/demo is ready.
Deployments stabilized in 22.870941299s
```

With the same condition of no images in the docker daemon, from Cloud Code:
```
skaffold dev --filename skaffold.yaml --label ide=idea --label ideVersion=2019.3.1.0.0 --label ijPluginVersion=unknown --rpc-port 50052 --port-forward=true --status-check=true --enable-rpc=true
time="2020-03-17T15:08:19-04:00" level=warning msg="port 50052 for gRPC HTTP server already in use: using 50053 instead"
Listing files to watch...
 - gcr.io/wise-coyote-827/jib-demo
Generating tags...
 - gcr.io/wise-coyote-827/jib-demo -> gcr.io/wise-coyote-827/jib-demo:d0be6fa-dirty
Checking cache...
 - gcr.io/wise-coyote-827/jib-demo: Found Locally
Tags used in deployment:
 - gcr.io/wise-coyote-827/jib-demo -> gcr.io/wise-coyote-827/jib-demo:86ffb97a970f634f9d351bf122655f4df59f1f547ab38b1a29b6c50cd1a2efc4
   local images can't be referenced by digest. They are tagged and referenced by a unique ID instead
Starting deploy...
 - service/demo created
 - deployment.apps/demo created
 - secret/demo-postgresql created
 - service/demo-postgresql-headless created
 - service/demo-postgresql created
 - statefulset.apps/demo-postgresql created
Waiting for deployments to stabilize
 - default:deployment/demo Waiting for deployment "demo" rollout to finish: 0 of 1 updated replicas are available...
```

The interesting thing is.. Skaffold did find a docker image locally,  and if I run `docker images`, i see no such image.

I suppose it somehow found a wrong docker daemon to use? I do have 3 different docker daemons (docker machine on GCE, minikube, and docker for desktop)...

**Comment by saturnism** [comment]
While applying the `SKAFFOLD_VERBOSITY` env var, I discovered that the system env var has my `DOCKER_*` env vars that determines which Docker daemon to use.

Even though I already removed those configurations from `.zshrc`, they are still in `.bash_profile`.  Even though I launched IntelliJ from zsh, it still picked up env var from `.bash_profile`.

Removing the entries from `.bash_profile` solved the issue.

That being said, IntelliJ does allow you to pick a Docker Daemon.  Would it make sense for Cloud Code to respect that configuration?  Will move thread.
